src/student.js

```diff
@@ -0,0 +1,42 @@
+export default class Student {
+  constructor(details) {
+    const { firstName, lastName, gitHubUserName, email, cohortId } = details
+
+    this.firstName = firstName
+    this.lastName = lastName
+    this.gitHubUserName = gitHubUserName
+    this.email = email
+    this.cohortId = cohortId
+
+    for (const key in this.details) {
+      if (key === 'studentId') continue
+
+      const val = this.details[key]
+      if (typeof val === 'undefined') {
+        throw new Error('student missing details')
+      }
+    }
+
+    this.studentId = Student.nextId()
+  }
+
+  get details() {
+    return {
+      firstName: this.firstName,
+      lastName: this.lastName,
+      gitHubUserName: this.gitHubUserName,
+      email: this.email,
+      cohortId: this.cohortId,
+      studentId: this.studentId
+    }
+  }
+
+  static nextId() {
```

Comment:
static = function that's only available to the object's other methods, right?

Reply:
@Uprising5034 @AllyDouillette 

Use it sparingly and never for something that mutates data. Lots of wacky bugs can occur from over-using statics